machina/apps/forum_member/utils.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from datetime import timedelta
+
+from django.contrib.sites.models import Site
+from django.utils import timezone
+
+from machina.core.db.models import get_model
+from machina.core.loading import get_class
+
+
+Post = get_model('forum_conversation', 'Post')
+NotificationEmail = get_class('forum_member.emails', 'NotificationEmail')
+
+
+def send_notifications(interval, email_class=None, context=None):
+    """
+    Send notification on email to the user that subscribe on topics.
+    """
+    email_class = email_class or NotificationEmail
+    email = email_class()
+
+    if not context:
+        context = {}
+
+    time_ago = timezone.now() - timedelta(seconds=interval)
+    for post in Post.objects.filter(created__gt=time_ago):
```

Comment:
There is a problem here in that posts that have not yet been approved will be included in this queryset so subscribers will be notified of posts that they won't be able to read.

Also comparing the creation time means that if the script stops/starts for any reason, emails might not get sent or might get sent twice

Reply:
We've addressed this on our fork https://github.com/aptivate/django-machina/commit/6f82cfd5b1a81ea605d23ee19d0927451ae0f9fe